### PR TITLE
Proposal: OpenMRS Styleguide

### PR DESCRIPTION
**Pull request by internettrans**
I'm about to push two more PRs related to the particular parts of the styleguide, too

### CODE REVIEW

**Comment by internettrans** [comment]
Will merge if/when there are three approvals from @fatmali @angshu @ojwanganto, @brandones

File: text/0008-styleguide.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+- Having a single-sourced, shared styleguide allows designers and developers to "change once, see it everywhere."
+
+## Alternatives
+OpenMRS could choose not to have a styleguide. This has some obvious downsides as explained above, but is truly a simple option that remains
+flexible. Doing so calls on the distribution authors to come up with their own look and feel.
+
+OpenMRS could port [the old OpenMRS styleguide](https://demo.openmrs.org/openmrs/uicommons/styleGuide.page) into the new SPA and microfrontends.
+This would potentially provide an easier migration path for people using the old styleguide to switch their server rendered code to
+client rendered within the SPA. However, the old styleguide is not very well known and not used much or at all by many of the large distributions.
+Starting from scratch gives us a clean slate to make intentional and good decisions, for both the UX design and software development.
```
Comment (fatmali) [inline text/0008-styleguide.md:60]: Completely agree.

**Review by fatmali — COMMENTED** [review]
I approve

File: text/0008-styleguide.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+for more information.
+
+The OpenMRS styleguide could not expose any global css classes, but rely entirely on javascript components. Doing so would give the maintainers
+of the styleguide more control over the user experience. However, it also complicates the styleguide code substantially, and raises the barrier
+to entry for using the styleguide. CSS classes are ubiquitous to all frontend code, whereas javascript components often take some time to
+set up and get going in everyone's framework and build process.
+
+The OpenMRS styleguide could not expose javascript components, but only css. This option would ensure the highest level of compatibility
+with javascript frameworks and codebases, since javascript components are much harder than css to use in all situations. See
+[the styleguide components proposal](/textt/0007-styleguide.components.md) for more discussion about this.
```
Comment (brandones) [inline text/0008-styleguide.md]: typo "/textt/"

File: text/0008-styleguide.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Common practices (not enforced)
+- Use existing styleguide css and components whenever possible during feature development.
+- Do not add too many things to the styleguide that become dead over time. Prefer a small set of things that are actively maintained
+  and heavily used. For example, a Button component is small enough that it can be easily maintained and heavily used,
+  whereas a searchable, sortable, paginated, filterable Table component is likely too hard to make fully customizable in a way
```
Comment (brandones) [inline text/0008-styleguide.md:84]: I dunno, [DataTables](https://datatables.net/) are pretty fantastic...

**Comment by brandones** [comment]
To the extent that we do create our own style guide, this looks good to me.

**Review by brandones — APPROVED** [review]

File: text/0008-styleguide.md (hunk 3 above)
Comment (internettrans) [reply text/0008-styleguide.md:84]: Yeah if we ever get to the point where we need documentation as shown below, then we've built something pretty crazy 😄 

*The quantity of documentation for a Data Table*

datatables

File: text/0008-styleguide.md (hunk 2 above)
Comment (internettrans) [reply text/0008-styleguide.md]: 👍 fixed

**Comment by mseaton** [comment]
FYI - I think this proposal looks great and generally agree with it all as written

**Comment by cromwel** [comment]
I think the proposal sparks an interesting concept to the OMRS design. Most of what is suggested is highly recommended for consistency and usability purposes. That is great.

File: text/0008-styleguide.md (hunk 1 above)
Comment (ojwanganto) [reply text/0008-styleguide.md:60]: I agree too

**Comment by ojwanganto** [comment]
I approve

**Comment by internettrans** [comment]
With three approvals from microfrontends squad members, this is ready to be merged.

**Pull request merged**